custodial-copy-backend/src/main/scala/uk/gov/nationalarchives/custodialcopy/OcflService.scala

```diff
@@ -83,6 +91,28 @@ class OcflService(ocflRepository: OcflRepository, semaphore: Semaphore[IO]) {
       missingAndNonMissingObjects("changedObjects")
     )
   }
+
+  def getAllFilePathsOnAnObject(ioId: UUID): IO[List[String]] = IO.blocking {
+    val potentialOcflObject = Try(ocflRepository.getObject(ioId.toHeadVersion))
+
+    potentialOcflObject match {
+      case Success(ocflObject) =>
+        val objectVersionFileInList: List[OcflObjectVersionFile] = ocflObject.getFiles.asScala.toList
+        objectVersionFileInList.map(_.getPath)
+
+      case Failure(_: NotFoundException) => throw new Exception(s"Object id $ioId does not exist")
+      case Failure(coe: CorruptObjectException) =>
+        ocflRepository.purgeObject(ioId.toString)
+        throw new Exception(
```

Comment:
I know we've got thrown execptions elsewhere but it would be good not to have any other ones. I had a tinker with this and this works without having to use `Try` at all.

```scala
def getAllFilePathsOnAnObject(ioId: UUID): IO[List[String]] =
    IO.blocking(ocflRepository.getObject(ioId.toHeadVersion))
      .map(_.getFiles.asScala.toList.map(_.getPath))
      .handleErrorWith {
        case nfe: NotFoundException =>  IO.raiseError(new Exception(s"Object id $ioId does not exist"))
        case coe: CorruptObjectException => IO.blocking(ocflRepository.purgeObject(ioId.toString)) >>
          IO.raiseError(new Exception(
            s"Object $ioId is corrupt. The object has been purged and the error will be rethrown so the process can try again",
            coe
          ))
        case e: Throwable =>
          IO.raiseError(new Exception(
          s"'getObject' returned an unexpected error '$e' when called with object id $ioId"
        ))
    }
```